### PR TITLE
Switch from ONPREMISE to SELF_HOSTED on backend

### PR DESCRIPTION
**Pull request by chadwhitacre**
Part of https://github.com/getsentry/self-hosted/issues/796.

### ⚠️ MERGE THESE FIRST ⚠️

- [x] https://github.com/getsentry/getsentry/pull/6841
- [x] https://github.com/getsentry/single-tenant/pull/429

### CODE REVIEW

**Review by AniketDas-Tekky — APPROVED** [review]

**Review by evanh — APPROVED** [review]

**Comment by chadwhitacre** [comment]
Thanks @AniketDas-Tekky @evanh. My plan is to recruit approvals for https://github.com/getsentry/getsentry/pull/6841 and https://github.com/getsentry/single-tenant/pull/429 at Backend TSC tomorrow, and then move forward with this hopefully right after the call.

**Comment by chadwhitacre** [comment]
I added backcompat for the old config knob in a7acd796b11c7a2e1f95b5200ab9b980714ec7e7, to support single-tenant in particular.

**Review by evanh — APPROVED** [review]
LGTM

File: src/sentry/utils/settings.py [hunk 1]
```diff
@@ -70,3 +70,13 @@ def get_error_message(cls, dependency, package):
...
+
+
+def is_self_hosted():
+    # Backcompat for rename to support old consumers, particularly single-tenant.
+    from django.conf import settings
+
+    try:
+        return settings.SENTRY_SELF_HOSTED
+    except AttributeError:
+        return settings.SENTRY_ONPREMISE
```
Comment (wedamija) [inline src/sentry/utils/settings.py:82]: Nit: `return getattr(settings, 'SENTRY_SELF_HOSTED', 'SENTRY_ONPREMISE')`

**Review by wedamija — APPROVED** [review]

File: src/sentry/utils/settings.py (hunk 1 above)
Comment (chadwhitacre) [reply src/sentry/utils/settings.py:82]:
```
$ python3
Python 3.9.8 (main, Nov 10 2021, 06:03:50) 
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> class settings:
...   bar = 1
... 
>>> getattr(settings, 'foo', 'bar')
'bar'
>>>
```

Could do:

```
getattr(settings, 'foo', getattr(settings, 'bar'))
```

but is that really nicer? 🤔

Comment (wedamija) [reply src/sentry/utils/settings.py:82]: Oops yeah, I'm fine with it as is.

**Pull request merged**